modules/basegl/src/shadercomponents/segmentsurfacecomponent.cpp

```diff
@@ -0,0 +1,154 @@
+ *
+ * Inviwo - Interactive Visualization Workshop
+ *
+ * Copyright (c) 2022 Inviwo Foundation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#include <modules/basegl/shadercomponents/segmentsurfacecomponent.h>
+
+#include <modules/opengl/shader/shaderutils.h>
+
+namespace inviwo {
+
+SegmentSurfaceComponent::SegmentSurfaceComponent(VolumeInport& atlas)
+    : ShaderComponent{}
+    , name_("atlasboundary")
+    , useAtlasBoundary_{"useAtlasBoundary", "Render ISO using atlas boundary", false}
+    , applyBoundaryLight_{"applyBoundaryLight", "Apply lighting to boundary", true}
+    , showHighlighted_("showHighlightedIndices", "Show Highlighted", true,
+                       vec3(1.0f, 0.906f, 0.612f))
+    , showSelected_("showSelectedIndices", "Show Selected", true, vec3(1.0f, 0.769f, 0.247f))
+    , showFiltered_("showFilteredIndices", "Show Filtered", true, vec3(0.5f, 0.5f, 0.5f))
+    , nearestSampler_{}
+    , linearSampler_{}
+    , textureSpaceGradientSpacingScale_{"gradientScaling", "Scale gradient spacing", 1.0f, 0.0f,
+                                        5.0f}
+    , atlas_{atlas} {
+
+    useAtlasBoundary_.addProperties(applyBoundaryLight_, textureSpaceGradientSpacingScale_,
+                                    showHighlighted_, showFiltered_, showSelected_);
+
+    nearestSampler_.setFilterModeAll(InterpolationType::Nearest);
+    nearestSampler_.setWrapModeAll(Wrapping::Repeat);
+    linearSampler_.setFilterModeAll(InterpolationType::Linear);
+    linearSampler_.setWrapModeAll(Wrapping::Repeat);
```

Comment:
Should set these as constructor arguments above

Reply:
Add filter and wrap mode to this `SegmentSurfaceComponent(VolumeInport& atlas, InterpolationType type, Wrapping wrap)` or add these line to initializer list? Assume the former of the two.